package.json

```diff
@@ -40,6 +40,7 @@
         "karma-phantomjs-launcher": "1.0.4",
         "karma-sourcemap-loader": "0.3.7",
         "karma-webpack": "4.0.2",
+        "lodash": "^4.17.10",

```

Comment:
Any particular reason for this added dev dependency?

Reply:
`lodash.throttle` it seems, but i'm not sure if it's worth the whole dependency. Request animation frame would also work ehre.

Reply:
Ahh, i totally missed the file `TableResize.ts` as it didn't auto load the diff. Yep, I agree that we shouldn't take that dependency for throttle. And it is listed as dev dependency which is not the case I believe as its being used in the core code and not tests.

Reply:
Yea, based on what I was looking at it seems as though lodash is a pretty widely used and quality package. That was one of the reasons why I went with it for throttling. We also have dependencies that use it (looking in our yarn.lock). I could refactor it to remove the dependency and use requestAnimationFrame instead

Reply:
In general, what is your team philosophy around taking dependencies vs writing custom in-repo solutions. What are the main things you tend to focus on when deciding to take a new dependency or not (maintainability , quality, flexibility, etc)?

Reply:
Good question! I'll let @JiuqingSong chime in for that as he has more thoughts on it.

P.S: Jason, I think we should document somewhere on whats the philosophy of adding (non-dev) dependencies.